app/filters/contribution_type_filter.rb

```diff
@@ -0,0 +1,22 @@
+class ContributionTypeFilter
+  def self.options
+    { name: 'Contribution Types', filters: [
+      { id: 'ContributionType[Ask]', name: 'Ask' },
+      { id: 'ContributionType[Offer]', name: 'Offer' }
+    ]}
+  end
+  ALL_ALLOWED_TYPES = ['Ask', 'Offer'].freeze
+
+  attr_reader :parameters
+
+  def initialize(params)
+    @parameters = params
+  end
+
+  def scopes
+    classes = parameters.blank? ? ALL_ALLOWED_TYPES : parameters.keys
+    classes.intersection(ALL_ALLOWED_TYPES).map do |type|
+      type.constantize.matchable
```

Comment:
Is this right — that `matchable` is what we want here?

Reply:
yes. we'll need to add that scope to `CommunityResource` to return objects that are still published